front/lib/actions/mcp_metadata.ts

```diff
@@ -170,18 +183,31 @@ export const connectToMCPServer = async (
     }
     case "remoteMCPServerUrl": {
       const url = new URL(params.remoteMCPServerUrl);
-      const sseTransport = new SSEClientTransport(url);
-      await mcpClient.connect(sseTransport);
+
+      try {
+        const sseTransport = new SSEClientTransport(url);
+        await mcpClient.connect(sseTransport);
+      } catch (e: unknown) {
+        return new Err(
+          new Error("Error establishing connection to remote MCP server.")
+        );
+      }
       break;
     }
 
     case "localMCPServerId": {
-      const transport = new ClientSideRedisMCPTransport(auth, {
-        conversationId: params.conversationId,
-        mcpServerId: params.mcpServerId,
-        messageId: params.messageId,
-      });
-      await mcpClient.connect(transport);
+      try {
```

Comment:
We're not supposed to try catch here either if the functions we call are ours?

Reply:
yup transport creation is ours and cannot fail, but mcpClient.connect is not ours and can fail, so i handle that

moving the try around only the mcpClient.connect

Reply:
That's the spirit 👍